pandas/core/indexes/base.py

```diff
@@ -2369,11 +2365,10 @@ def union(self, other):
                                   stacklevel=3)
 
         # for subclasses
-        return self._wrap_union_result(other, result)
+        return self._wrap_setop_result(other, result)
 
-    def _wrap_union_result(self, other, result):
-        name = self.name if self.name == other.name else None
-        return self.__class__(result, name=name)
+    def _wrap_setop_result(self, other, result):
+        return self.__class__(result, name=get_op_result_name(self, other))
```

Comment:
use ``self._shallow_copy`` this propagates meta-data, allows you to remove some sublcass implementations of this

Reply:
thats fine